algopy_testing/README.md

```diff
@@ -1,6 +1,71 @@
-# Algorand Python Testing framework
+# Algorand Python Testing
 
-Test your Algorand Python smart contracts using your favorite python testing tools.
+`algorand-python-testing` is a supplementary package to [`algorand-python`](https://github.com/algorandfoundation/puya) that allows unit testing of smart contract code directly. It offers offline, fast, and reliable unit testing capabilities, providing a familiar _pythonic_ testing experience.
+
+### Purpose and Benefits
+
+-   **Fast and Reliable**: Test your `Algorand Python` smart contracts offline, ensuring quick and dependable results without the need for blockchain deployment.
+-   **Granular Testing**: Testing individual contract methods, mock and patch contract state, pinpoint and verify contract behavior.
+-   **Property Testing**: Verify the contract's correctness under a variety of randomized inputs and constraints.
+-   **Comprehensive**: Supports a wide range of testing scenarios and use cases.
+
+[Documentation](TODO) | [Algorand Python Documentation](https://algorandfoundation.github.io/puya/) | [Algorand Python Language guide](https://algorandfoundation.github.io/puya/language-guide.html)
+
+## 1. Quick start
+
+### Installation
+
+To get started with unit testing your `Algorand Python` smart contracts, install the package with pip:
+
+```bash
+pip install algopy-testing-python
+```
+
+### Testing your first contract
+
+Let's write a simple "Hello World" contract and test it using the `algopy-testing-python` framework.
+
+Assume the following starter contract available on all `AlgoKit Python` templates out of the box (create via `algokit init -t python`):
+
+```python
+from algopy import ARC4Contract, arc4
+
+class HelloWorld(ARC4Contract):
+    @arc4.abimethod()
+    def hello(self, name: arc4.String) -> arc4.String:
+        return "Hello, " + name
+```
+
+Now, let's create a test file `test_hello_world.py`:
+
+```python
+from algopy_testing import algopy_testing_context
+from contracts.hello_world import HelloWorld
+
+def test_hello_world():
+    with algopy_testing_context() as ctx:
```

Comment:
worth showing this or as a context that is passed into the test? I guess the nice thing about this is it's testing framework agnostic?